### PR TITLE
Readme update (fixes Issue #89)

### LINKED ISSUE #89: Update README file
**Describe the bug**
The README file needs some process and clarification updates.

**To Reproduce**
- Open the README file.
- Read it.
- Wonder why it's so hard to understand.

**Expected behavior**
- Open the README file.
- Read it.
- Understand rosetta-bitcoin better.

**Additional context**

### PR DESCRIPTION
**Pull request by racbc**
Fixes #89.

### Motivation
The README file was a bit hard to read and left information gaps.

### Solution
This README update improves readability and expands on the information already in the doc.

### Open questions
Is there anything else I should add?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -207,18 +199,12 @@ in recently processed blocks to save to disk.
...
-
-## Future Work
-* Publish benchamrks for sync speed, storage usage, and load testing
-* [Rosetta API `/mempool/transaction`](https://www.rosetta-api.org/docs/MempoolApi.html#mempooltransaction) implementation
-* Add CI test using `rosetta-cli` to run on each PR (likely on a regtest network)
-* Add performance mode to use unlimited RAM (implementation currently optimized to use <= 16 GB of RAM)
-* Support Multi-Sig Sends
+* `rosetta-cli check:data --configuration-file rosetta-cli-conf/testnet/config.json` - This command validates that the Data API information in the `testnet` network is correct. It also ensures that the implementation does not miss any balance-changing operations.
+* `rosetta-cli check:construction --configuration-file rosetta-cli-conf/testnet/config.json` - This command validates the blockchain’s construction, signing, and broadcasting.
+* `rosetta-cli check:data --configuration-file rosetta-cli-conf/mainnet/config.json` - This command validates that the Data API information in the `mainnet` network is correct. It also ensures that the implementation does not miss any balance-changing operations.
```
Comment (shrimalmadhur) [inline README.md:204]: Same changes as https://github.com/coinbase/rosetta-ethereum/pull/75

Comment (racbc) [reply README.md]: Updated in new commit.

File: README.md [hunk 2]
```diff
@@ -26,16 +26,44 @@ information [here](https://rosetta-api.org).
 
 ## Features
 * Rosetta API implementation (both Data API and Construction API)
-* UTXO cache for all accounts (accessible using `/account/balance`)
+* UTXO cache for all accounts (accessible using the Rosetta API `/account/balance` function)
```
Comment (shrimalmadhur) [inline README.md:29]: I don't think so you need to mention this function. It's an API, which you already added.

File: README.md [hunk 3]
```diff
@@ -205,20 +197,13 @@ in recently processed blocks to save to disk.
...
+* `rosetta-cli check:construction --configuration-file rosetta-cli-conf/testnet/config.json` - This command validates the Construction API implementation. It also verifies transaction construction, signing, and submissions to the `testnet` network.
+* `rosetta-cli check:data --configuration-file rosetta-cli-conf/mainnet/config.json` - This command validates that the Data API implementation is correct using the Bitcoin `mainnet` node. It also ensures that the implementation does not miss any balance-changing operations.
 
-_Please reach out on our [community](https://community.rosetta-api.org) if you want to tackle anything on this list!_
+## Issues
+Interested in helping fix issues in this repository? You can find to-dos in the [Issues](https://github.com/coinbase/rosetta-bitcoin/issues) section with the `help wanted` tag. Be sure to reach out on our [community](https://community.rosetta-api.org) before you tackle anything on this list.
```
Comment (shrimalmadhur) [inline README.md:206]: we can remove the `help wanted` here too.

File: README.md (hunk 2 above)
Comment (racbc) [reply README.md:29]: Fixing in next commit.

File: README.md (hunk 3 above)
Comment (racbc) [reply README.md]: Fixed in next commit.

**Review by shrimalmadhur — APPROVED** [review]

**Pull request merged**